src/components/CardList.jsx

```diff
@@ -5,6 +5,7 @@ function CardList() {
 
   const apiURL="https://rickandmortyapi.com/api/character";
   const [characters, setCharacters] = useState([]);
+  const [showMore, setShowMore] = useState(false);
```

Comment:
You have one showMore state for all your cards? What is the intention behind this? This changes the state for all cards even if it is only pressed on one card.